common/context.go

```diff
@@ -240,10 +266,8 @@ func (ctx *Context) InitializeContext() error {
 	return nil
 }
 
-func loadYamlConfig(config *Config, yamlReader io.Reader) error {
-	yamlBuffer := new(bytes.Buffer)
-	yamlBuffer.ReadFrom(yamlReader)
-	return yaml.Unmarshal(yamlBuffer.Bytes(), config)
+func loadYamlConfig(config *Config, muContents string) error {
```

Comment:
this should remain as an `io.Reader` to allow the function to be called streams rather than always assuming the entire object is in memory.